### PR TITLE
Update Camel K CRDs

### PR DESCRIPTION
**Pull request by squakez**
By matching `APIVersion` field with the ones provided by Camel K 1.0 installation, the issue #8465 seems to be fixed.

### CODE REVIEW

**Comment by claudio4j** [comment]
@squakez in the `tools/bin/commands/util/camel_k_funcs` there is the `CAMEL_K_OPERATOR_DEFAULT_VERSION="0.3.4"` it should be updated to the camel-k version `1.0.1`.

File: install/operator/pkg/generator/assets/install/cluster/camelk.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #
 # From `kamel install -o yaml`
 #
-- apiVersion: apiextensions.k8s.io/v1beta1
```
[the remark this answers is not in the document]
Comment (phantomjinx) [reply install/operator/pkg/generator/assets/install/cluster/camelk.yml:4]: @claudio4j 
My feeling is that we should only be installing the CRD in a non-OLM environment. The expectation should be that the CRD is not our's to install/control but something we request for install if not already. That's what OLM and the OperatorHub provide.

**Comment by claudio4j** [comment]
I agree with @phantomjinx about using OLM to install camel-k on OLM capable cluster. But for now, I suggest to merge this PR once @squakez fix the CRDs and later I can work on using OLM to install the camel-k resources as you suggested.

**Comment by squakez** [comment]
> @squakez in the `tools/bin/commands/util/camel_k_funcs` there is the `CAMEL_K_OPERATOR_DEFAULT_VERSION="0.3.4"` it should be updated to the camel-k version `1.0.1`.

Good catch @claudio4j

File: install/operator/pkg/generator/assets/install/cluster/camelk.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (squakez) [reply install/operator/pkg/generator/assets/install/cluster/camelk.yml:4]: > The right approach would be to use the official crd installed by camel-k. I ran it, you can replace the content of this camelk.yml for this https://gist.github.com/claudio4j/995d244f14f60590f3a80f34cf1e483e
> 
> You can run it yourself, download camel-k 1.0.1, install camel-k crd `kamel install --cluster-setup` as cluster admin, then `kamel install -o yaml` to output the CRDs, but keep only the **CustomResourceDefinition**

It was my first option but for some reason my output is empty: https://gist.github.com/squakez/26ce6e92dbb13965c9019f618d9cf964 - I'll try running your gist's content.

**Comment by squakez** [comment]
Please, @claudio4j have another look as I've fixed the suggestion you gave.
As for a general approach I do agree with @phantomjinx and @claudio4j to have this merged in order to fix the actual procedure and follow up with another issue to improve the installation procedure.

**Pull request merged**